integration/connector_test.go

```diff
@@ -0,0 +1,506 @@
+	suite.mqttClient.Disconnect(uint(suite.cfg.MqttQuiesceMs))
+}
+
+func TestConnectorSuite(t *testing.T) {
+	suite.Run(t, new(ConnectorSuite))
+}
+
+func (suite *ConnectorSuite) TestConnectionStatus() {
+	type connectionStatus struct {
+		ReadySince time.Time `json:"readySince"`
+		ReadyUntil time.Time `json:"readyUntil"`
+	}
+
+	timeout := time.Duration(suite.cfg.StatusTimeoutMs * int(time.Millisecond))
+	threshold := time.Now().Add(timeout)
+
+	for {
+		statusURL := fmt.Sprintf("%s/features/ConnectionStatus/properties/status", suite.thingURL)
+		body, err := suite.doRequest("GET", statusURL)
+		if err != nil {
+			if time.Now().Before(threshold) {
+				continue
+			}
+			suite.T().Errorf("connection status property not available: %v", err)
+			break
+		}
+
+		status := &connectionStatus{}
+		err = json.Unmarshal(body, status)
+		require.NoError(suite.T(), err, "connection status should be parsed")
+
+		suite.T().Logf("%+v", status)
+		suite.T().Logf("current time: %v", time.Now())
+
+		forever := time.Date(9999, time.December, 31, 23, 59, 59, 0, time.UTC)
+		if status.ReadyUntil != forever {
+			if time.Now().Before(threshold) {
+				continue
+			}
+			suite.T().Errorf("readyUntil should be %v", forever)
+			break
+		}
+
+		delta := int64(suite.cfg.TimeDeltaMs)
+		assert.Less(suite.T(), status.ReadySince.UnixMilli(), time.Now().UnixMilli()+delta, "readySince should be before current time")
+		break
+	}
+}
+
+func (suite *ConnectorSuite) TestCommand() {
+	ws, err := suite.newWSConnection()
+	require.NoError(suite.T(), err, "cannot create a websocket connection to the backend")
+	defer ws.Close()
+
+	commandResponseCh := make(chan error)
+
+	dittoHandler := func(requestID string, msg *protocol.Envelope) {
+		if msg.Path == fmt.Sprintf("/features/%s/inbox/messages/%s", featureID, commandName) {
+			value, ok := msg.Value.(string)
+			if !ok {
+				commandResponseCh <- fmt.Errorf("unexpected message payload: %v, %T\n", msg.Value, msg.Value)
+				return
+			}
+			responsePayload := fmt.Sprintf(responsePayloadTemplate, value)
+			response := things.NewMessage(model.NewNamespacedID(msg.Topic.Namespace, msg.Topic.EntityName))
+			// respond to the message by using the outbox
+			path := strings.Replace(msg.Path, "/inbox/", "/outbox/", 1)
+			responseMsg := response.Envelope(
+				protocol.WithCorrelationID(msg.Headers.CorrelationID()),
+				protocol.WithResponseRequired(false),
+				protocol.WithContentType("application/json")).
+				WithTopic(msg.Topic).
+				WithPath(path).
+				WithValue(responsePayload).
+				WithStatus(http.StatusOK)
+			if err := suite.dittoClient.Reply(requestID, responseMsg); err != nil {
+				commandResponseCh <- fmt.Errorf("failed to send response: %v\n", err)
+			}
+			commandResponseCh <- nil
+		}
+	}
+	suite.dittoClient.Subscribe(dittoHandler)
+	defer suite.dittoClient.Unsubscribe(dittoHandler)
+
+	namespace := model.NewNamespacedIDFrom(suite.thingCfg.DeviceID)
+	cmd := things.NewCommand(namespace)
```

Comment:
It's not a command but message has to be used. Switching to Ditto Go library, the code should become: `things.NewMessage(namespace).Feature(featureID).Inbox(commandName)...`